Cython/Includes/libcpp/string.pxd

```diff
@@ -178,6 +181,7 @@ cdef extern from "<string>" namespace "std" nogil:
         string& append(const char* s) except +
         string& append(const char* s, size_t n) except +
         string& append(size_t n, char c) except +
+        string_ref append[input_iterator](input_iterator, input_iterator) except +  # (6)
```

Comment:
What does the `#6` refer to here?

Reply:
This refers to the number of the overload in the C++ standard. You will notice that in my rewrite (which I have abandoned for the purposes of the rebase until a consensus is reached, as explained in the comment), I have consistently numbered all overloads according to the standard to ensure that they are implemented correctly and none are missing.

<img width="933" alt="Screenshot 2025-01-21 at 19 10 49" src="https://github.com/user-attachments/assets/17c236e1-3089-4ec9-a931-1919398ab554" />